### PR TITLE
LPD-38057 Send the type property to the FieldBase since we can have the date or date_time types

### PR DESCRIPTION
**Pull request by rebsilva**
Related issue: https://liferay.atlassian.net/browse/LPD-38057

### CODE REVIEW

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:forms

**Comment by rebsilva** [comment]
ci:test:data-engine

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by rebsilva** [comment]
the unit test "DatePicker renders the label" is failing, i'm investigating why this is happening

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:forms

**Comment by rebsilva** [comment]
ci:test:data-engine

**Comment by rebsilva** [comment]
@cgmonte could you please review the unit test please? thanks! [d7cbfb3](https://github.com/liferay-forms/liferay-portal/pull/2633/commits/d7cbfb372b7bbd0d86b4232943aa2361cd9ace3d)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -40,8 +40,9 @@ describe('DatePicker', () => {
 		render(<DatePicker label="Date picker" />);
 
 		const allByText = screen.getAllByText('Date picker');
-		expect(allByText).toHaveLength(1);
+		expect(allByText).toHaveLength(2);
```
Comment (cgmonte) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:43]: why do we have two now? :thinking:

**Review by cgmonte — COMMENTED** [review]
Hmm...

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js (hunk 1 above)
Comment (cgmonte) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:43]: Is it because of the hidden one?

Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:43]: yes, i actually came back to what was before selena sent the fix, the test was exactly like this

if you see the tests from the other field types they are doing the exactly same thing

Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:43]: it's a span hidden that is used for accessibility

Comment (cgmonte) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:43]: So it LGTM! Just forward ir, if failures are not related.

Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/DatePicker/DatePicker.es.js:43]: thanks!

**Comment by rebsilva** [comment]
ci:forward

**Pull request closed without merging**